mailbox/jpa/src/main/java/org/apache/james/mailbox/jpa/user/JPASubscriptionMapper.java

```diff
@@ -48,11 +50,18 @@ public JPASubscriptionMapper(EntityManagerFactory entityManagerFactory) {
 
     @Override
     public void save(Subscription subscription) throws SubscriptionException {
+        EntityManager entityManager = getEntityManager();
+        EntityTransaction transaction = entityManager.getTransaction();
```

Comment:
what about using `TransactionRunner` instead?

Reply:
Thought about it.

But not doable as `getEntityManager()` is scoped to a mapper transaction and thus is larger than this mapper operation.

Reply:
Longs things short: JPA mappers are crap!

Reply:
You gonna be happy to know the mapper handles the transactions for us BTW...